### PR TITLE
RDCC-1902: Fixing NPE when file is not present

### PR DESCRIPTION
**Pull request by AdilHMCTS**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-1902

### Change description ###

Fixing NPE when file is not present

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by shashank-rastogi — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/data/ingestion/camel/processor/HeaderValidationProcessor.java [hunk 1]
```diff
@@ -35,6 +37,12 @@ public void process(Exchange exchange) throws Exception {
 
         RouteProperties routeProperties = (RouteProperties) exchange.getIn().getHeader(MappingConstants.ROUTE_DETAILS);
         String csv = exchange.getIn().getBody(String.class);
+
+        if (ofNullable(csv).isEmpty()) {
+            throw new RouteFailedException(
+                    routeProperties.getFileName() + " file is not found. Please upload file and try again.");
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/data/ingestion/camel/processor/HeaderValidationProcessor.java]: Use concat rather +?

Comment (AdilHMCTS) [reply src/main/java/uk/gov/hmcts/reform/data/ingestion/camel/processor/HeaderValidationProcessor.java]: This has been addressed here: https://github.com/hmcts/data-ingestion-lib/pull/10/commits/72e09b4ab03583c76c477a1bff7349d5c5defd1b

_However, it is good to note that `.concat()` will throw an NPE if `routeProperties.getFileName()` returns null_

**Review by abhijitdiwan — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -21,7 +21,7 @@ plugins {
     id 'com.github.ben-manes.versions' version '0.20.0'
     id "info.solidsoft.pitest" version '1.3.0'
     id 'io.spring.dependency-management' version '1.0.9.RELEASE'
-    id 'org.owasp.dependencycheck' version '5.3.0'
+    id 'org.owasp.dependencycheck' version '6.0.2'
     id 'org.sonarqube' version '2.8'
```
Comment (shashank-rastogi) [inline build.gradle:24]: do we need dependency check jar here? we shud be using hmcts gradle plugin

Comment (AdilHMCTS) [reply build.gradle:24]: Seems we already had the hmcts plugin, so I have updated it to the latest here: https://github.com/hmcts/data-ingestion-lib/pull/10/commits/515a6de4cd6777da0c8a781d50ebd84d51fe7307 and removed dependencycheck here: https://github.com/hmcts/data-ingestion-lib/pull/10/commits/bf60b8b312a7846299c90b58e9581a220c0bbcc2

**Review by shashank-rastogi — APPROVED** [review]

**Pull request closed without merging**